engine/collection/compliance/engine.go

```diff
@@ -216,14 +215,27 @@ func (e *Engine) OnSyncedClusterBlock(syncedBlock flow.Slashable[messages.Cluste
 func (e *Engine) finalizationProcessingLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
 	ready()
 
+	finalView := uint64(0)
+	var finalHeader *flow.Header
+
 	doneSignal := ctx.Done()
-	blockFinalizedSignal := e.finalizationEventsNotifier.Channel()
 	for {
 		select {
 		case <-doneSignal:
 			return
-		case <-blockFinalizedSignal:
-			e.core.ProcessFinalizedView(e.finalizedView.Value())
+		case finalizedBlock := <-e.finalizedBlocksNotifications:
+			// drop outdated notifications
+			if finalizedBlock.View < finalView {
```

Comment:
1. `finalView` is not updated.
2. Why do we need this? To prevent delivery out of order? This shouldn't happen at all, meaning we don't expect to receive height `n+1` before height `n`

Reply:
It's replicating what `if e.finalizedView.Set(block.View)` was doing previously. The only real purpose is to skip processing duplicate notifications, rather than out of order. I'd be happy to get rid of it though - dupe notifications are unlikely and their impact is low.

Reply:
Yeah, same here, I would get rid of it, makes logic simpler and it's very unlikely that this will happen.

Reply:
Removed in https://github.com/onflow/flow-go/pull/3598/commits/36c3380942411252725382b95820f5f3375b6d1c